### PR TITLE
driver/Keysight Infiniium oscilloscope (as e.g. MSOS104A)

### LINKED ISSUE #701: Keysight Infiniium S series scope drivers
We need drivers 
For the new blue force fridges
* MSOS254A http://www.keysight.com/en/pdx-x202069-pn-DSOS254A/high-definition-oscilloscope-25-ghz-4-analog-channels?nid=-32531.1150280&cc=DK&lc=dan
To aid the development of the mDAC  
* MSOS104A http://www.keysight.com/en/pdx-x202068-pn-DSOS104A/high-definition-oscilloscope-1-ghz-4-analog-channels?nid=-32531.1150279&cc=DK&lc=dan

### PR DESCRIPTION
**Pull request by Dominik-Vogel**
This is the driver for the Keysight Infiniium oscilloscopes.  It has been developed for the MSOS104A. So far it features all the basic commands, including:
 - time scale
 - channels
 - trigger
 - acquisition of up to 500M points (non streaming mode)

To get started please refer to the jupyter tutorial for this driver, supplied along with this PR.
fixes #701 
The driver development has been simultaneously started by Steven Waddy. In the first commit I merged the two drivers.

### CODE REVIEW

File: qcodes/instrument_drivers/Keysight/MSOS104A.py [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+        timestepsize = 1/srate
+        start = xorigin
+        stop = timestepsize*npts+xorigin
+        # what is the correct number of points?
+        npts = self.acquire_points.get_latest()
+
+        #
+        self.shape = (npts,)
+        self.set_sweep(start, stop, npts)
+        #TODO: fix above set when any of the above things change
```
Comment (jenshnielsen) [inline qcodes/instrument_drivers/Keysight/MSOS104A.py]: due to limitations in the dataset this cannot be set in the get command as the setpoints have already been written to fil

File: qcodes/instrument_drivers/Keysight/MSOS104A.py [hunk 2]
```diff
@@ -0,0 +1,368 @@
...
+                            channel = channel,
+                            parameter_class=RawTrace
+                            )
+        def snapshot_base(self, update: bool=False) -> Dict:
+            params_to_skip_update = ['trace']
+            super().snapshot_base(update=update,
+                                  params_to_skip_update=params_to_skip_update)
+
+
+class MSOS104A(VisaInstrument):
```
Comment (jenshnielsen) [inline qcodes/instrument_drivers/Keysight/MSOS104A.py]: I think it should have a more generic name the only difference that i can tell is the max frequency

File: qcodes/instrument_drivers/Keysight/MSOS104A.py (hunk 1 above)
Comment (Dominik-Vogel) [reply qcodes/instrument_drivers/Keysight/MSOS104A.py]: Thank you, this is a good hint!

**Comment by WilliamHPNielsen** [comment]
Nice notebook!

File: qcodes/instrument_drivers/Keysight/Infiniium.py [hunk 3]
```diff
@@ -0,0 +1,416 @@
...
+
+        # Acquisition
+        self.add_parameter(name='trace',
+                           channel=channel,
+                           parameter_class=RawTrace
+                           )
+
+        def snapshot_base(self, update: bool=False) -> Dict:
+            params_to_skip_update = ['trace']
+            super().snapshot_base(update=update,
```
Comment (jenshnielsen) [inline qcodes/instrument_drivers/Keysight/Infiniium.py]: This is missing a return statement

File: qcodes/instrument_drivers/Keysight/Infiniium.py [hunk 4]
```diff
@@ -0,0 +1,416 @@
...
+        executes command and sets trace_ready status to false
+        any command that effects the number of setpoints should invalidate the trace
+        """
+        self.trace_ready = False
+        self.write(cmd.format(val))
+
+
+# compatitbility/setup
+# change of names with acquire_ prefix
+# add acquire_interpolate(1) to the setup
```
Comment (jenshnielsen) [inline qcodes/instrument_drivers/Keysight/Infiniium.py]: Should these lines still be her?

File: qcodes/instrument_drivers/Keysight/Infiniium.py (hunk 3 above)
Comment (Dominik-Vogel) [reply qcodes/instrument_drivers/Keysight/Infiniium.py]: Thanks! I fixed this.

File: qcodes/instrument_drivers/Keysight/Infiniium.py (hunk 4 above)
Comment (Dominik-Vogel) [reply qcodes/instrument_drivers/Keysight/Infiniium.py]: nope...

**Pull request merged**